src/eigen/preconditioners.jl

```diff
@@ -66,5 +66,5 @@ end
 (Base.:*)(P::PreconditionerTPA, R) = mul!(copy(R), P, R)
 
 function precondprep!(P::PreconditionerTPA, X)
-    P.mean_kin = [real(dot(x, P.kin .* x)) for x in eachcol(X)]
+    P.mean_kin = [sum(real.(conj.(x) .* P.kin .* x)) for x in eachcol(X)]
```

Comment:
Why is that faster? `dot` should just be a BLAS call no?

Reply:
Putting the `real` part in should in theory allow more optimizations, and it's 2x less storage (real instead of cplx). It's just exposing to julia what it needs to be in theory faster, I did not benchmark to see if it was actually faster

Reply:
good point.